src/gui/folderwatcher_win.cpp

```diff
@@ -97,47 +97,47 @@ void WatcherThread::watchChanges(size_t fileNotifyBufferSize,
             break;
         }
         if (result != 0) {
-            qCWarning(lcFolderWatcher) << "WaitForMultipleObjects failed" << result << GetLastError();
+            qCWarning(lcFolderWatcher) << "WaitForMultipleObjects failed" << result << Utility::formatWinError();
             break;
         }
 
         bool ok = GetOverlappedResult(_directory, &overlapped, &dwBytesReturned, false);
         if (!ok) {
-            DWORD errorCode = GetLastError();
+            const DWORD errorCode = GetLastError();
             if (errorCode == ERROR_NOTIFY_ENUM_DIR) {
                 qCDebug(lcFolderWatcher) << "The buffer for changes overflowed! Triggering a generic change and resizing";
                 emit lostChanges();
                 emit changed(_path);
                 *increaseBufferSize = true;
             } else {
-                qCWarning(lcFolderWatcher) << "GetOverlappedResult error" << errorCode;
+                qCWarning(lcFolderWatcher) << "GetOverlappedResult error" << Utility::formatWinError(errorCode);
             }
             break;
         }
 
         FILE_NOTIFY_INFORMATION *curEntry = pFileNotifyBuffer;
         forever {
-            size_t len = curEntry->FileNameLength / 2;
-            QString file = _path + "\\" + QString::fromWCharArray(curEntry->FileName, len);
+            const int len = curEntry->FileNameLength / sizeof(wchar_t);
+            QString longfile = longPath + QString::fromWCharArray(curEntry->FileName, len);
 
             // Unless the file was removed or renamed, get its full long name
             // TODO: We could still try expanding the path in the tricky cases...
-            QString longfile = file;
             if (curEntry->Action != FILE_ACTION_REMOVED
                 && curEntry->Action != FILE_ACTION_RENAMED_OLD_NAME) {
-                size_t longNameSize = GetLongPathNameW(reinterpret_cast<LPCWSTR>(file.utf16()), fileNameBuffer, fileNameBufferSize);
+                const auto wfile = longfile.toStdWString();
```

Comment:
yes toStdWString creates a copy and as the code is windows only the result is supposed to be the same, but ....